cargo-concordium/src/main.rs

```diff
@@ -187,61 +196,84 @@ enum Command {
         about = "Build a deployment ready smart-contract module."
     )]
     Build {
-        #[structopt(
-            name = "schema-embed",
-            long = "schema-embed",
-            short = "e",
-            help = "Builds the contract schema and embeds it into the wasm module."
-        )]
-        schema_embed:      bool,
-        #[structopt(
-            name = "schema-out",
-            long = "schema-out",
-            short = "s",
-            help = "Builds the contract schema and writes it to file at specified location."
-        )]
-        schema_out:        Option<PathBuf>,
-        #[structopt(
-            name = "schema-json-out",
-            long = "schema-json-out",
-            short = "j",
-            help = "Builds the contract schema and writes it in JSON format to the specified \
-                    directory."
-        )]
-        schema_json_out:   Option<PathBuf>,
-        #[structopt(
-            name = "schema-base64-out",
-            long = "schema-base64-out",
-            short = "b",
-            help = "Builds the contract schema and writes it in base64 format to file at \
-                    specified location or prints the base64 schema to the console if the value \
-                    `-` is used. The path has to exist while the file will be created. (expected \
-                    input: `./my/path/base64_schema.b64` or `-`)."
-        )]
-        schema_base64_out: Option<PathBuf>,
-        #[structopt(
-            name = "out",
-            long = "out",
-            short = "o",
-            help = "Writes the resulting module to file at specified location."
-        )]
-        out:               Option<PathBuf>,
-        #[structopt(
-            name = "contract-version",
-            long = "contract-version",
-            short = "v",
-            help = "Build a module of the given version.",
-            default_value = "V1"
-        )]
-        version:           utils::WasmVersion,
-        #[structopt(
-            raw = true,
-            help = "Extra arguments passed to `cargo build` when building Wasm module."
-        )]
-        cargo_args:        Vec<String>,
+        #[structopt(flatten)]
+        build_options: BuildOptions,
     },
 }
 
+// TODO: A doc comment here replaces the `about` string in Test and Build.
```

Comment:
Can't we add an explicit `about` where it is used then?

What other options should we consider here?

Reply:
There is already an explicit `about`, but this seems to take precedence. I'll look into other options.

Reply:
Without doc comments on `BuildOptions`:
<img width="704" alt="image" src="https://github.com/Concordium/concordium-smart-contract-tools/assets/5469652/0f932e2e-7e5a-469b-9a28-7fddeeea2166">

With doc comments:
<img width="581" alt="image" src="https://github.com/Concordium/concordium-smart-contract-tools/assets/5469652/e64c0ac5-8be5-4c60-9d10-6049c2b48a01">

Reply:
I see. This macro is a bit too magical :) 

I guess we should then just have a comment to make sure somebody does not add a doc comment in the future :)

Reply:
Yep 🪄🐇